TodoApp/src/TodoApp.Api/Controllers/ItemsController.cs

```diff
@@ -69,7 +66,7 @@ public async Task<IHttpActionResult> PostAsync(Item item)
 
         public async Task<IHttpActionResult> PutAsync(Guid id, Item item)
```

Comment:
This method is really simple right now and doesn't calculate with some behaviour:
- you should handle item validity and return BadRequest if not (you have it in POST, why not here?)
- you should handle id emptiness and create new Item (like in POST) if id is empty
- you should handle nonexisting id (return NotFound)

Reply:
Also, you should call itemUpdater here from the controller (same way you call itemCreator in POST).

Reply:
Also, check item existency in Delete method and after deletion clear the cache ;-)